wagtail/test/utils/wagtail_tests.py

```diff
@@ -1,12 +1,43 @@
 import warnings
 from contextlib import contextmanager
 
+from bs4 import BeautifulSoup
 from django import VERSION as DJANGO_VERSION
 from django.contrib.auth import get_user_model
+from django.test.client import Client as DjangoTestClient
 from django.test.testcases import assert_and_parse_html
 
 
+class BS4SoupClient(DjangoTestClient):
+    """Convenience wrapper around Django's Client and BeautifulSoup's SoupSieve
+
+    The intend of this wrapper is to make it easier to check the html of a response.
+
+    Usage example:
+
+        client = BS4SoupClient()
+        response = client.get("/good-vibes/")
+
+        # Select a HTML <span class="good-vibes> in the response
+        good_vibes = response.select("span.good-vibes")
+
+        # Assert the is only one element on the page that matches the selector
+        self.assertEqual(len(soup), 1)
+    """
+
+    def request(self, **request):
+        response = super().request(**request)
+        response.soup = BeautifulSoup(response.content, "html5lib")
+
+        # Alias these BeautifulSoup functions the the response for convenience
+        response.select = response.soup.select
+        response.select_one = response.soup.select_one
+        return response
+
+
 class WagtailTestUtils:
+    client_class = BS4SoupClient
```

Comment:
I see one problem with this: all test classes that extend `WagtailTestUtils` will run BeautifulSoup and parse the HTML every time a client request is made, even if the test case doesn't need it. I think this needs to be made optional somehow, e.g. using a flag on the class or making it so that BeautifulSoup is only run once.

Reply:
I raised similar in https://github.com/wagtail/wagtail/pull/10708#discussion_r1269691585.

Perhaps it needs to be a method, `get_soup`, that does the parsing easily. That way it's up to the test writer to actually use it. A `cached_property` would be nicer, but munging that on to the request is a bit weirder.

Reply:
Addressed this, see my comment: https://github.com/wagtail/wagtail/pull/10708#discussion_r1276253701